internal/bminventory/inventory.go

```diff
@@ -304,6 +306,35 @@ func (b *bareMetalInventory) updatePullSecret(pullSecret string, log logrus.Fiel
 	return pullSecret, nil
 }
 
+func getDefaultHighAvailabilityAndMasterCountParams(highAvailabilityMode *string, controlPlaneCount *int64) (*string, *int64) {
+	// Both not set, multi node by default
+	if highAvailabilityMode == nil && controlPlaneCount == nil {
+		return swag.String(models.ClusterCreateParamsHighAvailabilityModeFull),
+			swag.Int64(common.MinMasterHostsNeededForInstallationInHaMode)
+	}
+
+	// only highAvailabilityMode set
+	if controlPlaneCount == nil {
+		if *highAvailabilityMode == models.ClusterHighAvailabilityModeNone {
```

Comment:
here you are checking if the Masters are schedulable 
I think you should use / modify the current function for it instead of re writing something new - also this need to be applied to it as well casue we do use it in a lot of places
https://github.com/openshift/assisted-service/blob/master/internal/common/common.go#L193